realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -281,6 +342,41 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread util transactions are committed by other threads to this Realm or
+     * {@link #stopWaitForChange()} gets called.
+     *
+     * @return {@code true} if transactions are committed to this Realm, {@code false} if
+     * {@link #stopWaitForChange()} gets called.
+     * @throws IllegalStateException if attempting to wait within a transaction.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        sharedGroupManager.setWaitForChangeEnabled(true);
+        // add this to wait list for batch stop
+        addRealmToWaitList(this);
+        boolean hasChanged = sharedGroupManager.waitForChange();
+        // remove this from wait list
+        removeRealmFromWaitList(this);
+        if (hasChanged) {
+            sharedGroupManager.advanceRead();
```

Comment:
IMO `waitForChange` shouldn't work on a Looper thread at all since the looper should control all communication and Realm refresh. At least that was the idea with introducing this delayed notification concept and deprecating `refresh`, and since async doesn't work on non-looper threads it is a lot easier.